doc: update multisig tutorial to use multipath descriptors

BenWestgate opened the pull request:
### Summary

Update `doc/multisig-tutorial.md` to use multipath descriptor format  
instead of separate external/internal descriptors. The tutorial now:

- extracts a single `xpub_n` per participant
- constructs a multipath `wsh(sortedmulti(...))` descriptor with `<0;1>`  
  change index semantics
- uses `getdescriptorinfo` to compute descriptor checksum
- explains that `importdescriptors` expands the multipath descriptor  
  into internal and external descriptors
- update `/test/functional/wallet_multisig_descriptor_psbt.py` functional test / documentation to use multi-path descriptors

---

### Motivation

A single multipath descriptor is the most convenient pattern for multisig; our documentation should use it.

---

### What changed

- replaced extraction of `external_xpub_n` and `internal_xpub_n` with  
  extraction of a single `xpub_n`
- removed instructions to create and import separate external/internal  
  descriptors
- added instructions to build a multipath `wsh(sortedmulti(...))`  
  descriptor and derive checksum with `getdescriptorinfo`
- checksum field is parsed and appended as the multipath descriptor is not the canonical "desc" output
- clarified that `importdescriptors` automatically expands multipath  
  descriptors into internal and external forms
- similar changes to the functional test: wallet_multisig_descriptor_psbt.

---

### Testing

I have run the updated shell snippets and confirmed the multipath descriptor produces the same `listdescriptors` output after importing as the two descriptor method in bitcoin:master.

---

### Related issues / PRs

This tutorial change references the multipath descriptor
consolidation (see commit / PR referenced in the change). The commit
message points to bitcoin#22838 as the upstream change that enables
this behavior.

---

### Release note (for changelog)

Documentation: update multisig tutorial and multisig functional test to use multipath descriptors

BenWestgate committed:
doc: Update multisig-tutorial.md to use multipath descriptors

Update doc/multisig-tutorial.md to use a single multipath descriptor
instead of separate external/internal descriptors, per PR #22838.
Extract one xpub per participant, build a multipath descriptor with
<0;1> change index, and use getdescriptorinfo to append the checksum.
Clarify importdescriptors expands multipath descriptors into internal
and external forms. Tested shell snippets to confirm equivalent
listdescriptors output as the two-descriptor method.

Added missing loadwallet command for multisig_wallet_01

test: Use multipath descriptors in the functional wallet test
wallet_multisig_descriptor_psbt as this is intended as documentation

doc: replace `bitcoin-cli` with `bitcoin rpc` in multisig-tutorial.md

removed -named parameter where possible.

fixed a couple bugs where -signet was not passed

the call to getcoins.py requires the bitcoin-cli command still

Sjors commented:
Concept ACK

#32784 contains a similar change, but it's nice to do that separately and earlier.

rkrux commented on doc/multisig-tutorial.md:
```diff
@@ -65,46 +63,43 @@ for x in "${!xpubs[@]}"; do printf "[%s]=%s\n" "$x" "${xpubs[$x]}" ; done
+`desc` specifies the output type (`wsh`, in this case) and the xpubs involved. It also uses BIP 67 (`sortedmulti`), so the wallet can be recreated without worrying about the order of xpubs. Conceptually, descriptors describe a list of scriptPubKey (along with information for spending from it) [[source](https://github.com/bitcoin/bitcoin/issues/21199#issuecomment-780772418)].
 
-Note that at least two descriptors are usually used, one for internal derivation paths and one for external ones. There are discussions about eliminating this redundancy, as can be seen in the issue [#17190](https://github.com/bitcoin/bitcoin/issues/17190).
+Note that previously at least two descriptors were usually used, one for internal derivation paths and one for external ones. Since https://github.com/bitcoin/bitcoin/pull/22838 this redundancy has been eliminated by a multipath descriptor with <code><0;1></code> as the change index.
```
In cf2986684ba0db4e0f7b32deb65d7767f80e9c01 "Update multisig-tutorial.md to use multipath descriptors"

Can consider moving this note in the previous section where the multipath descriptor is actually sed-ed.
```diff
diff --git a/doc/multisig-tutorial.md b/doc/multisig-tutorial.md
index fa6366df91..c59359de0a 100644
--- a/doc/multisig-tutorial.md
+++ b/doc/multisig-tutorial.md
@@ -63,6 +63,8 @@ for x in "${!xpubs[@]}"; do printf "[%s]=%s\n" "$x" "${xpubs[$x]}" ; done
 
 As previously mentioned, this step extracts the `m/84'/1'/0'` account instead of the path defined in [BIP 45](https://github.com/bitcoin/bips/blob/master/bip-0045.mediawiki) or [BIP 87](https://github.com/bitcoin/bips/blob/master/bip-0087.mediawiki), since there is no way to extract a specific path in Bitcoin Core at the time of writing.
 
+Note that previously at least two descriptors were usually used, one for internal derivation paths and one for external ones. Since https://github.com/bitcoin/bitcoin/pull/22838 this redundancy has been eliminated by a multipath descriptor with <code><0;1></code> as the change index.
+
 ### 1.2 Define the Multisig Descriptor
 
 Define the multisig descriptor, add the checksum and then, wrap it in a JSON array.
@@ -77,8 +79,6 @@ multisig_desc="[{\"desc\": \"${desc}#${checksum}\", \"active\": true, \"timestam
 
 `desc` specifies the output type (`wsh`, in this case) and the xpubs involved. It also uses BIP 67 (`sortedmulti`), so the wallet can be recreated without worrying about the order of xpubs. Conceptually, descriptors describe a list of scriptPubKey (along with information for spending from it) [[source](https://github.com/bitcoin/bitcoin/issues/21199#issuecomment-780772418)].
 
-Note that previously at least two descriptors were usually used, one for internal derivation paths and one for external ones. Since https://github.com/bitcoin/bitcoin/pull/22838 this redundancy has been eliminated by a multipath descriptor with <code><0;1></code> as the change index.
-
 After creating the descriptor, it is necessary to add the checksum, which is required by the `importdescriptors` RPC.
 
 The checksum for a descriptor without one can be computed using the `getdescriptorinfo` RPC. The response has the `descriptor` field, which is the descriptor with the checksum added.
```

rkrux reviewed (commented):
Concept ACK 226836e70501abf9ac2434900e74eec31edd716e

As mentioned in this comment https://github.com/bitcoin/bitcoin/pull/33286#issuecomment-3248584420 as well, I think all these commits can be squashed into one.

BenWestgate commented:
Thanks for the suggestion @Sjors. I've updated the test to use multi-path descriptors as well.

> Although not directly related to this PR, I have a minor suggestion: it might be helpful to add a loadwallet command for multisig_wallet_01, not just for participant wallets.

I agree it's a flaw to not load multisig_wallet_01 so I added it to this PR, we'll see if that's OK or reviewers want it in a separate commit or PR.

Sjors commented on doc/multisig-tutorial.md:
````diff
@@ -120,6 +115,7 @@ Once the wallets have already been created and this tutorial needs to be repeate
 
 ```bash
 for ((n=1;n<=3;n++)); do ./build/bin/bitcoin-cli -signet loadwallet "participant_${n}"; done
+./build/bin/bitcoin-cli -signet loadwallet "multisig_wallet_01"
````
nit: you could use `./build/bin/bitcoin rpc` instead. This removes the need for `-named` elsewhere in the tutorial.

Sjors reviewed (approved):
utACK 5424b4f1ce74c82b7ae01034bbc1592088048128

I didn't test the tutorial.

kannapoix commented on doc/multisig-tutorial.md:
````diff
@@ -183,9 +177,9 @@ Optionally, the PSBT can be decoded to a JSON format using `decodepsbt` RPC.
 
 ```bash
-./build/bin/bitcoin-cli -signet decodepsbt $funded_psbt
+./build/bin/bitcoin rpc -signet decodepsbt $funded_psbt
````
This command may result in an error like:
```
error code: -8
error message:
Unknown named parameter cHNidP8BAH0CAAAAASMcBn5VJaIET6lL4b76V676ffyrvpxT6UI++gY14LwWAQAAAAD9////AoAaBgAAAAAAFgAUhv4Mbo6zEszAaF4zBry18DXCw+CuhQEAAAAAACIAIKUhPbGKzw2gMuSI/kFtVa5EYyTBZdlS7zeuyJg5kJ2kAAAAAAABAH0CAAAAARULOYOh2W+IbDULcj5n6NF4uzQx8ZcqXa+YrNgx+Y/XmwMAAAD9////AqQGAAAAAAAAFgAUvI6wMmXzdKiS7/nhen+pYfbaY8MgoQcAAAAAACIAILvTfOv022A80QUkWRjFA2e0RkHffMZf5J8TeDWo3ajRIB8EAAEBKyChBwAAAAAAIgAgu9N86/TbYDzRBSRZGMUDZ7RGQd98xl/knxN4NajdqNEBBWlSIQJJlgJ1txTezfeaWmWQOhvtX8QpacAatIPHwclsdFKSZSECnQKWtfbPyvYEekbMRgQjGQHLD4vD/Cknn12bTH1WFRkhA7q6xvV1rI/SuSmR/3VAoEd6wymtY0S3GOyp+dNgZhSSU64iBgJJlgJ1txTezfeaWmWQOhvtX8QpacAatIPHwclsdFKSZRgdWhVBVAAAgAEAAIAAAACAAAAAAAAAAAAiBgKdApa19s/K9gR6RsxGBCMZAcsPi8P8KSefXZtMfVYVGRhsho9WVAAAgAEAAIAAAACAAAAAAAAAAAAiBgO6usb1dayP0rkpkf91QKBHesMprWNEtxjsqfnTYGYUkhh47RKgVAAAgAEAAIAAAACAAAAAAAAAAAAAAAEBaVIhArIbxpgn95IGTodGk2VVW6kpXgW6sI7rnSuJeYyeSpAKIQMkA2pAaaug/2ZQvQspY1KMldrpV2E38WBI5gk/+rZsmiEDJbyWjQ7C1EB74/9f2H140782VImwBhoB9OyC0hUQGRdTriICArIbxpgn95IGTodGk2VVW6kpXgW6sI7rnSuJeYyeSpAKGHjtEqBUAACAAQAAgAAAAIABAAAAAAAAACICAyQDakBpq6D/ZlC9CyljUoyV2ulXYTfxYEjmCT/6tmyaGB1aFUFUAACAAQAAgAAAAIABAAAAAAAAACICAyW8lo0OwtRAe+P/X9h9eNO/NlSJsAYaAfTsgtIVEBkXGGyGj1ZUAACAAQAAgAAAAIABAAAAAAAAAAA
```

IIUC, you may need to use ‎`-nonamed` when passing parameters that include ‎`=`.
A PSBT may contain ‎`=` as base64 padding.

https://github.com/bitcoin/bitcoin/blob/3d42607fb5966d8e3b79fdb878d59483432625d9/src/bitcoin.cpp#L86-L91

```suggestion
./build/bin/bitcoin rpc -nonamed -signet decodepsbt $funded_psbt
```

We also need to apply the same fix as for `decodepsbt` to `analyzepsbt`, `walletprocesspsbt`, `combinepsbt`, and `finalizepsbt`.

Note: Quoting a PSBT didn't solve the issue.
```
./build/bin/bitcoin rpc -signet decodepsbt "$funded_psbt"
```

rkrux commented on test/functional/wallet_multisig_descriptor_psbt.py:
```diff
@@ -44,26 +45,19 @@ def _check_psbt(psbt, to, value, multisig):
             result = multisig.importdescriptors([
-                {  # receiving addresses (internal: False)
-                    "desc": external["descriptor"],
+                {  # Multipath descriptor expands to change and receive
```
In 5424b4f1ce74c82b7ae01034bbc1592088048128 "doc: Update multisig-tutorial.md to use multipath descriptors"

Nit: By convention, the first index of the two corresponds to the receive address and the other being the change address as quoted in the above tutorial too: `doc/multisig-tutorial.md`.

```diff
- # Multipath descriptor expands to change and receive
+ # Multipath descriptor expands to receive and change 
```

rkrux commented on doc/multisig-tutorial.md:
```diff
@@ -44,14 +44,14 @@ wpkh([1004658e/84'/1'/0']tpubDCBEcmVKbfC9KfdydyLbJ2gfNL88grZu1XcWSW9ytTM6fitvaRm
 The suffix (after #) is the checksum. Descriptors can optionally be suffixed with a checksum to protect against typos or copy-paste errors.
 All RPCs in Bitcoin Core will include the checksum in their output.
 
+Note that previously at least two descriptors were usually used, one for internal derivation paths and one for external ones. Since https://github.com/bitcoin/bitcoin/pull/22838 this redundancy has been eliminated by a multipath descriptor with <code><0;1></code> as the change index.
```
In 5424b4f1ce74c82b7ae01034bbc1592088048128 "doc: Update multisig-tutorial.md to use multipath descriptors"

>  with <code><0;1></code> as the change index.

I find it confusing to read it as the change index when one of the indices is non-change (receive) when expanded later. Maybe the below one?

```diff
- with <code><0;1></code> as the change index.
+ with <code><0;1></code> as the derivation index. 
```

----------------------------------------
Nit: To keep it consistent with how the multipath is expanded.
```diff
- , one for internal derivation paths and one for external ones.
+ , one for external derivation paths and one for internal ones. 
```

BenWestgate replied:
I updated the note with your feedback and implemented the comment nit in the test.  Does this address the confusion?

> Note that previously at least two descriptors were usually used, one for external derivation paths and one for internal ones. Since https://github.com/bitcoin/bitcoin/pull/22838 this redundancy has been eliminated by a multipath descriptor with <code><0;1></code> at the [BIP-44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#change) change level expanding to external and internal descriptors when imported.

The term used in BIP-44 is "change level" which is why I used that word to match that standard, but I agree it's confusion without reference that the change level determines whether the chain is internal or external.

rkrux reviewed (commented):
crACK 3d42607fb5966d8e3b79fdb878d59483432625d9

rkrux reviewed (approved):
crACK 2a46e94

BenWestgate committed:
doc: Update add checksum instructions in tutorial

Clarify how to add the checksum for a descriptor in non-canonical form.

rkrux reviewed (approved):
crACK de7c3587cd4586bbed94a4ea6eae4a252301daee

Sorry for missing out on the question earlier, I verified it using the `getdescriptorinfo` RPC for a multipath descriptor and it looks good to me.

Sjors commented:
utACK de7c3587cd4586bbed94a4ea6eae4a252301daee

BenWestgate commented:
> utACK [de7c358](https://github.com/bitcoin/bitcoin/commit/de7c3587cd4586bbed94a4ea6eae4a252301daee)

This commit is documentation only so our tutorial text matches what we do in shell snippets.

Tests have not changed since 2a46e94a1600a4f28e01db23a89f039acaa2c45e on Oct 6.

kannapoix commented:
Core review ACK: de7c3587cd4586bbed94a4ea6eae4a252301daee

achow101 commented:
ACK de7c3587cd4586bbed94a4ea6eae4a252301daee
